packages/weak-store/README.md

```diff
@@ -0,0 +1,3 @@
+# WeakStore
+
+Wrapper for JavaScript WeakMap
```

Comment:
what does it add? why not just use `WeakMap`?

Reply:
Mark prefers using this for two reasons:

1) It distinguishes between initializing (`init`) a key, value pair and reseting the key to a different value (`set`), whereas `WeakMap` doesn't. This means you can use this abstraction and not have to check whether the key already exists - the method that you call marks your intention and does it for you.  

2) You can use the methods in a functional programming pattern, which you can't with WeakMap. For instance, you can create a new function `const getPurse = weakStore.get` and you can do `myArray.map(weakStore.get)`. You can't do this with WeakMap. I think Mark said it's because the method is not tied to a particular WeakMap instance. Here's what happens when you try it:

```
> const wm = new WeakMap();
undefined
> const a = {};
undefined
> wm.set(a, 1);
WeakMap { <items unknown> }
> wm.get(a)
1
> const getSomething = wm.get;
undefined
> getSomething(a)
Thrown:
TypeError: Method WeakMap.prototype.get called on incompatible receiver undefined
    at get (<anonymous>)
```

Reply:
Ah, ok, so `Store` is a name for a new kind of Map-like object, and thus `WeakStore` is the corresponding weak form. Ok, in that case `store` and `weak-store` really are the best names, and we just need to (eventually) improve the README to explain what these things are and how they differ from the native Map objects.

Reply:
yes

Reply:
Just added the above explanation to the READMEs for Store and WeakStore